### PR TITLE
Adding initial documentation

### PR DESCRIPTION
**Pull request by aakarshg**
Adding initial documentation

### CODE REVIEW

**Comment by aakarshg** [comment]
@jtaleric @portante @chaitanyaenr @ndokos @bengland2 PTAL

File: README.md [hunk 1]
```diff
@@ -1,2 +1,29 @@
 # scribe
-Project to implement the Red hat Performance and Scale data model. This project will depend on data collection from Stockpile.
+
+Project to implement the Red hat Performance and Scale data model.
+This project will depend on data collection from Stockpile.
+
+## Functionality of Scribe
+
+The main Functionality of scribe is to provide the indexer with files, that
+have the user requested data and ensuringg that it conforms to CDM
```
Comment (chaitanyaenr) [inline README.md]: s/ensuringg/ensuring

Comment (chaitanyaenr) [inline README.md]: s/CDM/Common Data Model

File: README.md [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
+
+## How to run scribe
+
+Currently scribe takes 3 arguments:
+
+1. input_path ( `--input-path` or `-ip` ): This is the path where stockpile's
+generated machine_facts.json file is located. Default is '/tmp/machine_facts.json'
+
+2. output_path( `--output-path` or `-ip` ): This is the path where scribe will
```
Comment (chaitanyaenr) [inline README.md]: s/ip/op

**Review by chaitanyaenr — CHANGES_REQUESTED** [review]

File: scribe.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+#             list_keys_cdm.append(previous_key)
+
+
+# def get_keys(dl, keys_list):
+#     if isinstance(dl, dict):
+#         keys_list += dl.keys()
+#         map(lambda x: get_keys(x, keys_list), dl.values())
+#     elif isinstance(dl, list):
+#         map(lambda x: get_keys(x, keys_list), dl)
+
```
Comment (jtaleric) [inline scribe.py:32]: Remove all commented out code prior to merging PR

**Review by jtaleric — COMMENTED** [review]
Camel Case with underscores? I would pick one or the other, I would personally go with all lower case with underscores. 

Given scribe is going to be adapted, why not take a OO approach? 

ScribeBase, which gets inherited by all additional scribe plugins. For example, we have stockpile data for Packages -- and this should be consistent across all scribe data. I would anticipate we would want to have a Scribe Packages class. 

Packages(ScribeBase) for example. 

Thoughts / overkill for the initial kickoff?

**Comment by aakarshg** [comment]
Yes, I wanted to separate functions and variables, and didn't want to lose using underscores. I updated them to all lower case with underscores. 

As much as I want scribe to adapt OO approach, adding a scribe class for packages will mean that it's dependent on stockpile, knows of its existence and based on conversation with peter, that's something we don't want to do. I guess there could be a way to look at CDM and figure out what the variable is and build the class but yeah probably overkill for now as we are yet to come together on CDM

**Comment by jtaleric** [comment]
Why not take the OO approach to have a "Stockpile" plugin then? That is predicated on the fact the input is from Stockpile?

**Pull request closed without merging**